lua/wire/sh_wirenet.lua

```diff
@@ -0,0 +1,210 @@
+--[[
+	Consolidated networking library for Wiremod.
+	Trivial Networking - A wrapper for sending very small net messages that don't need a whole networkstring.
+]]
+
+local Net = WireLib.Net or {}
+WireLib.Net = Net
+
+-- Trivial Networking
+local SIZE = 10 -- 1023 names
+
+local registered_handlers = {}
+
+local update_handlers
+
+local function msg_handler(len, ply)
+	local handler = registered_handlers[net.ReadUInt(SIZE)]
+	if handler then
+		handler(len, ply)
+	else
+		error("WireLib.Net tried to receive message that is not registered.")
+	end
+end
+
+net.Receive("wirelib_net_message", msg_handler)
+
+if SERVER then
+	util.AddNetworkString("wirelib_net_message")
+
+	local MAXLEN = 4096
+
+	local handler_names = {}
+	local handler_queue = {}
+	local queue_handler_firstidx, queue_handler_lastidx
+	local function queue_handler_flush(ply, tbl, first, last)
```

Comment:
Same review as sparky, we should just use netstreams instead since advdupe already uses it and it's better to have a standardized system.